Example/Tests/SHA256Tests.swift

```diff
@@ -0,0 +1,34 @@
+//
+//  SHA256Tests.swift
+//  Krypt_Tests
+//
+//  Created by Simon Feistel on 21.01.20.
+//  Copyright © 2020 CocoaPods. All rights reserved.
+//
+
+import Krypt
+import XCTest
+
+class SHA256Tests: XCTestCase {
+  func testSAH256Calculation_nonBuffered__shouldProduceCorrectHash() {
+    // given
+    let testData = TestData.largeTestData
+    /// Was calculated locally using openssl
+    let expectedHash = "4b9ecf8a918594442fa56d14f2f8e975b258d134bf71c3a05c913b952d40af46"
+    // when
+    let digest = SHA256.digest(testData.data)
+    // then
+    XCTAssertEqual(digest, Data(hex: expectedHash))
+  }
+
+  func testSHA256Calculation_buffered__shouldProduceSameAsUnbuffered() {
+    // given
+    /// Test data large enough to force at least 5 buffer fills
+    let testData = TestData.largeTestData
+    // when
+    let unbufferedDigest = SHA256.digest(testData.data)
+    let bufferedDigest = try! SHA256.digest(file: testData.url)
+    // then
+    XCTAssertEqual(unbufferedDigest, bufferedDigest)
+  }
```

Comment:
Don't like that the digest from file test is using the digest from data to produce the expected hash, it's not really a unit test at this point.

```swift
class SHA256Tests: XCTestCase {

  // Computed using `openssl dgst -sha256 -hex [file path]
  let expectedHash = "4b9ecf8a918594442fa56d14f2f8e975b258d134bf71c3a05c913b952d40af46"

    func testDigestData__shouldProduceExpectedHash() {
      // given
      let testData = TestData.largeTestData

      // when
      let digest = SHA256.digest(testData.data)

      // then
      XCTAssertEqual(digest, Data(hex: expectedHash))
    }
 
    func testDigestFile_whenTestDataLargeEnoughFor5BufferFills__shouldProduceExpectedHash() {
      // given
      let testData = TestData.largeTestData

      // when
      let digest = try! SHA256.digest(file: testData.url)

      // then
      XCTAssertEqual(digest, Data(hex: expectedHash))
    }
```
Can we do something like this?